server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceConfig.java

```diff
@@ -68,15 +68,22 @@
         Object missing,
         ZoneId timeZone,
         String format,
-        Function<Script, ValuesSourceType> resolveScriptAny
-    ) {
+        Function<Script, ValuesSourceType> resolveScriptAny,
+        String aggregationName) {
+        ValuesSourceConfig<VS> config;
 
         if (field == null) {
+            // Stand Alone Script Case
             if (script == null) {
-                ValuesSourceConfig<VS> config = new ValuesSourceConfig<>(ValuesSourceType.ANY);
-                config.format(resolveFormat(null, valueType, timeZone));
-                return config;
+                throw new IllegalStateException(
+                    "value source config is invalid; must have either a field context or a script or marked as unwrapped");
             }
+            /*
+             * This is the Stand Alone Script path.  We should have a script that will produce a value independent of the presence or
+             * absence of any one field.  The type of the script is given by the valueType field, and defaults to bytes if not specified.
+             */
+            // TODO: Not pluggable, should always be valueType if specified, BYTES if not.
+            // TODO: Probably should validate that the resulting type is valid for this agg.  That needs to be plugable.
             ValuesSourceType valuesSourceType = valueType != null ? valueType.getValuesSourceType() : ValuesSourceType.ANY;
```

Comment:
Worth noting that by the time we get here, `valueType` is hopelessly muddled.  It could be user-specified, or it could be a default value from the aggregation builder, or it could be null.  We have no way of knowing the source at this point.  I have a plan to clean that up, but in the interest of small, manageable PRs, I want to do that in another PR.